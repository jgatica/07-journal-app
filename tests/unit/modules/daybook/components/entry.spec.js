import {shallowMount} from "@vue/test-utils";
import Entry from "@/modules/daybook/components/Entry.vue";
import journalState from "../../../mock-data/test-journal-state";

describe('Pruebas componente Entry', () => {

  let wrapper

  const mockRouter = {
    push: jest.fn()
  }

  beforeEach(()=>{
    wrapper = shallowMount(Entry, {
      props: {
        entry: journalState.entries[0]
      },
      global: {
        mocks: {
          $router: mockRouter,
        }
      }
    })
  })


  it('debe hacer match con SnapShot ', () => {

    expect(wrapper.html()).toMatchSnapshot()
  });

  it('debe redireccionar al hacer click en el entry-container', async () => {
    const resultContenedor = wrapper.find('.entry-container')

    await resultContenedor.trigger('click')

    expect(mockRouter.push).toHaveBeenCalled()
    expect(mockRouter.push).toHaveBeenCalledWith(
      {
        name: 'entry',
        params: {
          id: journalState.entries[0].id
        }
      }
    )

  });

  it('pruebas en las propiedades computadas ', () => {
    expect(wrapper.vm.day).toBe(16)
    expect(wrapper.vm.month).toBe('Diciembre')
    expect(wrapper.vm.yearDay).toBe('2023, SÃ¡bado')
    console.log(wrapper.vm.day, wrapper.vm.month, wrapper.vm.yearDay)
  });

});