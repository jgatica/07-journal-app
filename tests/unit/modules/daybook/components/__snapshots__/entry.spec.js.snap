// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente Entry debe hacer match con SnapShot  1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">16</span><span class="mx-1 fs-5">Diciembre</span><span class="mx-2 fw-light">2023, SÃ¡bado</span></div>
  <div class="entry-description">Ejemplo 1</div>
</div>
`;
